xarray/backends/netCDF4_.py

```diff
@@ -206,10 +215,24 @@ def __init__(self, filename, mode='r', format='NETCDF4', group=None,
         self.ds = opener()
         self.format = format
         self.is_remote = is_remote_uri(filename)
-        self._opener = opener
         self._filename = filename
         self._mode = 'a' if mode == 'w' else mode
+        self._opener = functools.partial(_open_netcdf4_group, filename,
```

Comment:
Changed to `self._opener = functools.partial(opener, mode=self._mode)`.  Is this the best way to do this from previous usage of 
```python
  self._opener = opener(mode=self._mode)
                                         mode=self._mode,
                                         group=group, clobber=clobber,
                                         diskless=diskless, persist=persist,
                                         format=format)
```
?